catalog/data_storage_credential.go

```diff
@@ -0,0 +1,24 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+)
+
+func DataSourceStorageCredential() common.Resource {
+	type AccountMetastoreByID struct {
+		Name              string                         `json:"name"`
+		StorageCredential *catalog.StorageCredentialInfo `json:"storage_credential_info,omitempty" tf:"computed" `
+	}
+	return common.WorkspaceData(func(ctx context.Context, data *AccountMetastoreByID, w *databricks.WorkspaceClient) error {
```

Comment:
Workspace-level only because structures aren't compatible?

Reply:
yes, plus this makes it available for users who are not account admins (but have the necessary permissions to use storage credentials)

Reply:
Ok, maybe we can add support for account-level later (in addition to workspace-level)